src/LogicTasks/Syntax/SimplestFormula.hs

```diff
@@ -135,20 +143,32 @@ partialGrade' SuperfluousBracketsInst{..} f
     correctAtoms = sort $ nub $ collectLeaves tree
     correctOpsNum = numOfOps tree
 
-completeGrade :: OutputCapable m => SuperfluousBracketsInst -> Delayed FormulaAnswer -> LangM m
+completeGrade :: (OutputCapable m, Alternative m, Monad m) => SuperfluousBracketsInst -> Delayed FormulaAnswer -> Rated m
 completeGrade inst = completeGrade' inst `withDelayedSucceeding` parser
 
-completeGrade' :: OutputCapable m => SuperfluousBracketsInst -> FormulaAnswer -> LangM m
+completeGrade' :: (OutputCapable m, Alternative m, Monad m) => SuperfluousBracketsInst -> FormulaAnswer -> Rated m
 completeGrade' inst sol
-    | show (fromJust (maybeForm sol)) /= simplestString inst = refuse $ do
-      instruct $ do
-        english "Your solution is incorrect."
-        german "Ihre Lösung ist falsch."
-
-      when (showSolution inst) $ do
-        example (simplestString inst) $ do
-          english "The solution for this task is:"
-          german "Die Lösung für diese Aufgabe ist:"
+  | show sol == simplestString inst = rate 1
+  | synTreeEq = reRefuse (rate percentage) (translate $ do
+    german ("Sie haben " ++ show superfluousBracketsSubmission ++ " überflüssige Klammer(n) in der Abgabe.")
+    english ("You left " ++ show superfluousBracketsSubmission ++ " superfluous bracket(s) in your submission."))
+  | otherwise = reRefuse (rate 0) (translate $ do
+    german "Sie haben die Formel verändert."
+    english "You changed the formula.")
 
-      pure ()
-    | otherwise = pure()
+  where
+    countBrackets :: String -> Integer
+    countBrackets = fromIntegral . length . filter (== '(')
+    submission = fromJust (maybeForm sol)
+    synTreeSubmission = toSynTree submission
+    bracketsSubmission = countBrackets $ show submission
+    bracketsSolution = countBrackets $ simplestString inst
+    bracketsMax = countBrackets $ stringWithSuperfluousBrackets inst
+    superfluousBracketsSubmission = bracketsSubmission - bracketsSolution
+    superfluousBracketsSolution = bracketsMax - bracketsSolution
```

Comment:
Ist der Name hier wirklich angebracht? Es ist ja nicht die Anzahl der überflüssigen Klammern in der Lösung (so wie der Wert darüber tatsächlich die Anzahl der überflüssigen Klammern in der Einreichung ist).